generators/dart/math.js

```diff
@@ -204,35 +244,25 @@ Blockly.Dart['math_number_property'] = function(block) {
          '  }',
          '  return true;',
          '}']);
-    code = functionName + '(' + number_to_check + ')';
-    return [code, Blockly.Dart.ORDER_UNARY_POSTFIX];
-  }
-  switch (dropdown_property) {
-    case 'EVEN':
-      code = number_to_check + ' % 2 == 0';
-      break;
-    case 'ODD':
-      code = number_to_check + ' % 2 == 1';
-      break;
-    case 'WHOLE':
-      code = number_to_check + ' % 1 == 0';
-      break;
-    case 'POSITIVE':
-      code = number_to_check + ' > 0';
-      break;
-    case 'NEGATIVE':
-      code = number_to_check + ' < 0';
-      break;
-    case 'DIVISIBLE_BY':
-      const divisor = Blockly.Dart.valueToCode(block, 'DIVISOR',
-          Blockly.Dart.ORDER_MULTIPLICATIVE);
-      if (!divisor) {
-        return ['false', Blockly.Dart.ORDER_ATOMIC];
-      }
-      code = number_to_check + ' % ' + divisor + ' == 0';
-      break;
+
+    code = functionName + '(' + numberToCheck + ')';
+
+  } else if (dropdownProperty == 'DIVISIBLE_BY') {
+
+    const divisor = Blockly.Dart.valueToCode(block, 'DIVISOR',
+
+        Blockly.Dart.ORDER_MULTIPLICATIVE) || '0';
+
```

Comment:
To be consistent with other language generators, we'll want to keep the `|| '0'` above, but we'll add the following:
```suggestion

    if (divisor === '0') {
      return ['false', Blockly.Dart.ORDER_ATOMIC];
    }
```